code/modules/mob/living/carbon/life.dm

```diff
@@ -303,21 +303,26 @@
 
 	return TRUE
 
-//Fourth and final link in a breath chain
+/// Fourth and final link in a breath chain
 /mob/living/carbon/proc/handle_breath_temperature(datum/gas_mixture/breath)
 	// The air you breathe out should match your body temperature
 	breath.temperature = bodytemperature
 
+/// Attempts to take a breath from the external or internal air tank.
 /mob/living/carbon/proc/get_breath_from_internal(volume_needed)
-	if(internal)
-		if(internal.loc != src)
-			internal = null
-		else if ((!wear_mask || !(wear_mask.clothing_flags & MASKINTERNALS)) && !getorganslot(ORGAN_SLOT_BREATHING_TUBE))
-			internal = null
-		else
-			. = internal.remove_air_volume(volume_needed)
-			if(!.)
-				return FALSE //to differentiate between no internals and active, but empty internals
+	if(invalid_internals())
+		// Unexpectely lost breathing apparatus and ability to breathe from the internal air tank.
+		cutoff_internals()
+		return
+	if (external)
+		. = external.remove_air_volume(volume_needed)
+	else if (internal)
+		. = internal.remove_air_volume(volume_needed)
```

Comment:
why not just return here?

Reply:
I tried early returns and it mucks the logic up. Like this:
```
if (external)
    return external.remove_air_volume(volume_needed)
else if (internal)
    return internal.remove_air_volume(volume_needed)
```
This causes a bug that allows you to breathe from your turf despite being connected to an open, but empty tank. The below conditional was definitely required. Maybe there is a simpler way to write it, but I couldn't think of one.
```
// To differentiate between no internals and active, but empty internals.
    return . || FALSE
```